src/Disyuntor.ts

```diff
@@ -156,15 +155,19 @@ export function wrapCallbackApi<T extends (...args: any[]) => void>(
     disyuntor.protect(() => {
       return new Promise((resolve, reject) => {
         const newArgs = args.slice(0, -1)
-          .concat((err?: Error, ...args: any[]) => {
+          .concat((err: Error | null, ...cbArgs: any[]) => {
             if (err) { return reject(err); }
-            resolve(...args);
+            resolve(cbArgs);
           });
         call.call(thisParam, ...newArgs)
       });
     }).then(
-      (...args) => {
-        callback(null, ...args)
+      (args) => {
+        if (Array.isArray(args)) {
+          callback(null, ...args);
+        } else {
+          callback(null, args);
+        }
```

Comment:
I don't particularly love this if/else.  Anyone have some syntactic sugar that might do this for me?

Reply:
not sure this is any any better 🤷 

```javascript
callback(null, ...(Array.isArray(args) ? args : [args]))
```

Reply:
I don't know about the syntax sugar but what we don't want is to unexpectedly spread the arguments .. Let's say that the function we are "protecting" does something like this
```js
function test(callback) {
 setImmediate( () => callback(null, [1,2,3] ) );
}
```
should preserve the array. Excuse me is this is obvious but with so many `...` is hard to follow 🤣

Reply:
Good call out @jfromaniello !  I had *not* considered this case explicitly.

However, I just wrote a test using something like the case you gave here... and I believe it to be working.  Here's why: when we hit the Array.isArray(args) check, we actually have a single-element array (containing the multi-value array we passed to the callback) like this:

![Screen Shot 2021-02-04 at 3 32 09 PM](https://user-images.githubusercontent.com/2659880/106957848-71c8ac80-6730-11eb-92cb-ac1be22c98dd.png)

At this point I'm not certain that we'll ever need the `else` block though 🤔